src/Perlang.Tests.Integration/EvalHelper.cs

```diff
@@ -268,32 +268,32 @@ internal static object EvalWithArguments(string source, params string[] argument
 
         private static void AssertFailScanErrorHandler(ScanError scanError)
         {
-            throw scanError;
+            throw new Exception("ScanError occurred. See inner exception for details.", scanError);
         }
 
         private static void AssertFailParseErrorHandler(ParseError parseError)
         {
-            throw parseError;
+            throw new Exception("ParseError occurred. See inner exception for details.", parseError);
         }
 
         private static void AssertFailNameResolutionErrorHandler(NameResolutionError nameResolutionError)
         {
-            throw nameResolutionError;
+            throw new Exception("NameResolutionError occurred. See inner exception for details.", nameResolutionError);
         }
 
         private static void AssertFailRuntimeErrorHandler(RuntimeError runtimeError)
         {
-            throw runtimeError;
+            throw new Exception("RuntimeError occurred. See inner exception for details.", runtimeError);
         }
 
         private static void AssertFailValidationErrorHandler(ValidationError validationError)
         {
-            throw validationError;
+            throw new Exception("ValidationError occurred. See inner exception for details.", validationError);
         }
 
         private static bool AssertFailCompilerWarningHandler(CompilerWarning compilerWarning)
         {
-            throw compilerWarning;
+            throw new Exception("CompilerWarning occurred. See inner exception for details.", compilerWarning);
```

Comment:
The previous approach (creating an exception in one place, throwing it elsewhere) had one major drawback: it completely destroys the debugging experience, since the stack trace gets generated _when the exception is thrown_. Hence, it's much better to use the `InnerException` approach for this.

(#232 was similar to this, but the problem at that time was re-throwing new exceptions without providing the inner exception details; they were silently discarded, also negatively affecting the debugging experience.)